fe/fe-core/src/main/java/com/starrocks/catalog/ExprRangePartitionInfo.java

```diff
@@ -0,0 +1,197 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.catalog;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Range;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.analysis.CastExpr;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.FunctionCallExpr;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.util.RangeUtils;
+import com.starrocks.persist.gson.GsonUtils;
+import com.starrocks.server.RunMode;
+import com.starrocks.sql.analyzer.AnalyzerUtils;
+import com.starrocks.sql.analyzer.PartitionExprAnalyzer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * ExprRangePartitionInfo is an enhanced version of ExpressionRangePartitionInfo
+ * because ExpressionRangePartitionInfo is not easily scalable
+ * and get more extensions by extracting objects
+ * in the future this will replace all expr range partition info
+ */
+public class ExprRangePartitionInfo extends RangePartitionInfo {
+
+    private static final Logger LOG = LogManager.getLogger(ExprRangePartitionInfo.class);
+
+    @SerializedName(value = "partitionExprs")
```

Comment:
Do not serialize Expr directly, this will affect the unified Json of Expression serialized in the future